File: 0110-balanced-binary-tree/0110-balanced-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool res = true;
    bool isBalanced(TreeNode* root) {
        dfs(root);
        return res;
    }

    int dfs(TreeNode* node){
        if(!node) return 0;
        int lst = dfs(node->left);
        int rst = dfs(node->right); 
        if(abs(lst-rst)>1) res=false;
        return 1+max(lst,rst);
    }
};

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __init__(self): 
        self.res = True 

    def dfs(self,node): 
        if node is None: 
            return 0 
        lst = self.dfs(node.left) 
        rst = self.dfs(node.right)
        if abs(lst-rst)>1: 
            self.res=False 
        return 1+max(lst,rst) 

    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        if root is None: 
            return True 
        self.dfs(root)
        return self.res
        